org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/NullAnnotationModelTests.java

```diff
@@ -140,12 +141,19 @@ public void testConvertedSourceType1() throws CoreException, InterruptedExceptio
 
 			getCompilationUnit("/P/p2/C2.java").getWorkingCopy(this.wcOwner, null);
 
-			assertProblems("Unexpected problems", "----------\n" +
-					"1. WARNING in /P/p2/C2.java (at line 8)\n" +
-					"	return arg == null ? null : arg.toString();\n" +
-					"	       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n" +
-					"Null type safety: The expression of type 'String' needs unchecked conversion to conform to \'@NonNull String\'\n" +
-					"----------\n");
+			assertProblems("Unexpected problems", """
+					----------
+					1. ERROR in /P/p2/C2.java (at line 8)
+						return arg == null ? null : arg.toString();
+						                     ^^^^
+					Null type mismatch: required '@NonNull String' but the provided value is null
+					----------
+					2. WARNING in /P/p2/C2.java (at line 8)
+						return arg == null ? null : arg.toString();
+						                            ^^^^^^^^^^^^^^
+					Null type safety (type annotations): The expression of type 'String' needs unchecked conversion to conform to '@NonNull String'
+					----------
+					""");
```

Comment:
Sorry for slow reaction.
This change looks good, considering that type resolution for ternary expressions (`?:`) has changed in 1.8.

Reply:
Not sure we are on same page: I meant the test fail in [NullAnnotationModelTests/testBug565246/](https://ci.eclipse.org/jdt/job/eclipse.jdt.core-Github/job/PR-2551/37/testReport/junit/org.eclipse.jdt.core.tests.model/NullAnnotationModelTests/testBug565246/).
Currently I've simply disabled test.

Reply:
I'm interested in all changes in files like this. This was the first interesting change I found.

And after our discussion about the other test I even found your comment here :)